### DISCUSSION TITLE
Certificates of immunity

### DISCUSSION THREAD

**Discussion by colmjude** [digital-land/data-standards-backlog#93] [Planning consideration]
# Certificates of immunity

Certificates of immunity are a guarantee that a building or buildings will not be newly listed or given a building preservation notice for the next 5 years

## Data on planning.data.gov.uk

As of summer '24 we have data from Historic England (HE) on planning.data.gov.uk. The [dataset](https://www.planning.data.gov.uk/dataset/certificate-of-immunity) we are collecting is point data and has ~160 records.

**Comment by Swati-Dash** [comment]
COI Polygon data is available on HE website https://opendata-historicengland.hub.arcgis.com/datasets/historicengland::national-heritage-list-for-england-nhle/about?layer=5. We currently have COI point data on our platform. Can this polygon data be added as well? If we add Polygon data, do we still need to keep the point data?

**Reply by colmjude** [reply]
Hi @Swati-Dash, thanks for pointing this out.
It sounds like adding polygon data would be useful for certificates of immunity.
In terms of whether we need to keep the point data too - so keep collecting the dataset we are currently collecting - we'll need to do some analysis of the datasets to understand if there any differences and what they are.
We will add this to our backlog.

**Reply by colmjude** [reply]
## Analysis of dataset

We've done some initial analysis of this dataset.

### Folder Contents

The following files were found in the folder:

- Certificate_of_Immunity_polygons.cpg: Size: 5 bytes
- Certificate_of_Immunity_polygons.dbf: Size: 151029 bytes
- Certificate_of_Immunity_polygons.prj: Size: 417 bytes
- Certificate_of_Immunity_polygons.shp: Size: 173300 bytes
- Certificate_of_Immunity_polygons.shp.xml: Size: 455 bytes
- Certificate_of_Immunity_polygons.shx: Size: 1500 bytes

Total Folder Size: 326,706 bytes

### Summary

Geometry Type: MultiPolygon, Polygon
Number of Records: 175
Features/Attributes: ListEntry, Name, COIStart, COIExpire, CaptureSca, hyperlink, area_ha, NGR, Easting, Northing, geometry
Coordinate Reference System (CRS): EPSG:27700
File Size: 173300 bytes

### Sample

This example is for one of the certificates of immunity entities we already have point data for on planning.data.gov.uk (
[Cooling Towers at the former Willington Power Station [1420581]](https://www.planning.data.gov.uk/entity/2300000))

Sample GeoJSON

```
GeoJSON Sample Data
Sample GeoJSON content:
{
"type": "FeatureCollection",
"name": "national_heritage_sample",
"crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:EPSG::27700" } },
"features": [
{ "type": "Feature", "properties": { "ListEntry": 1420581, "Name": "Cooling Towers at the former Willington Power Station", "COIStart": "2020-01-13", "COIExpire": "2025-01-12", "CaptureSca": "1:2500", "hyperlink": "https://historicengland.org.uk/listing/the-list/list-entry/1420581", "area_ha": 1.84220842002, "NGR": "SK3108828910", "Easting": 431179.0, "Northing": 329046.0 }, "geometry": { "type": "MultiPolygon", "coordinates": [ [ [ [ 431179.740000000223517, 329011.370000001043081 ], [ 431176.160000000149012, 329011.370000001043081 ], [ 431173.810000000521541, 329011.670000001788139 ], [ 431172.900000001303852, 329011.770000003278255 ], [ 431170.450000000186265, 329012.169999999925494 ], [ 431168.0, 329012.869999999180436 ], [ 431167.180000000633299, 329013.170000003650784 ], [ 431164.940000000409782, 329013.98000000230968 ], [ 431162.690000001341105, 329014.990000002086163 ], [ 431161.78000000026077, 329015.490000000223517 ], [ 431159.74000000115484, 329016.720000002533197 ], [ 431157.709999999031425, 329018.050000000745058 ], [ 431156.790000000037253, 329018.660000002011657 ], [ 431153.340000000782311, 329021.940000001341105 ], [ 431152.629999999888241, 329022.650000000372529 ], [ 431149.600000000558794, 329026.540000000968575 ], [ 431149.090000000782311, 329027.359999999403954 ], [ 431146.669999999925494, 329031.640000002458692 ], [ 431146.480000001378357, 329032.560000004246831 ], [ 431145.589999999850988, 329034.799999998882413 ], [ 431144.920000000856817, 329037.140000002458692 ], [ 431144.73000000230968, 329038.150000000372529 ], [ 431144.280000001192093, 329040.48000000230968 ], [ 431144.030000002123415, 329042.920000001788139 ], [ 431144.049999999813735, 329043.929999999701977 ], [ 431143.900000002235174, 329046.369999999180436 ], [ 431144.150000001303852, 329048.800000002607703 ], [ 431144.2700
```

**Comment by Swati-Dash** [comment]
Thanks Colm. Are you suggesting to add polygon data? If we add Polygon data, do we still need to keep the point data?

**Reply by colmjude** [reply]
Hi @Swati-Dash 

The polygons should be addition to the entity and we can keep both the polygon and point so that users have the choice to use whichever works best for them.

For example, [entity 2300000](https://www.planning.data.gov.uk/entity/2300000) is available in the polygon dataset so we can collect this and add a geometry to this entity.

Of the 330 records on planning.data.gov.uk 161 overlap with the polygon dataset (see [coi_overlapping_records.csv](https://github.com/user-attachments/files/18555038/coi_overlapping_records.csv))

--
For reference
* [coi_non_overlapping_polygon_records.csv](https://github.com/user-attachments/files/18555041/coi_non_overlapping_polygon_records.csv)
* [coi_non_overlapping_platform_records.csv](https://github.com/user-attachments/files/18555042/coi_non_overlapping_platform_records.csv)

**Reply by colmjude** [reply]
Mappings for the polygon dataset

| field | mapped_from | notes |
| --- | --- | --- |
| `reference` | ListEntry | |
| `name` | Name | |
| `geometry` | | use the feature from the geojson |
| `point` | Easting, Northing | |
| `documentation-url` | NHLE link | |
| `entry-date` | | use the date first seen |
| `start-date` | COIStart | |
| `end-date` | COIExpire | |

**Comment by Swati-Dash** [comment]
Hi Colm, any update on this?

**Reply by colmjude** [reply]
Thanks for the nudge - see above
